# Informe: Sumativa 1 - Estadística Computacional
## Proyecto: Análisis integral del dataset meteorológico (weatherAUS)

**Asignatura:** Estadística Computacional  
**Profesor:** Jean Paul Maidana  
**Fecha:** 01 de julio de 2026  
**Institución:** Universidad Andrés Bello

---

### Integrantes del Grupo 7
 Nombres: 
| **Alana Bermúdez** |
| **Gonzalo Mansilla** |
| **Eduardo Villalón** |

---

# Preparación y Limpieza de Datos 
*Carga, limpieza y validación del dataset.*



## 1. Carga del Dataset
En esta etapa, importamos la librería *pandas* para procesar los datos estructurados.

In [122]:
import pandas as pd
df = pd.read_csv('weatherAUS.csv') # Asegúrate de que el nombre sea exacto
df.head() # Esto te mostrará las primeras filas de tu dataset

,Date,Location,MinTemp,MaxTemp,Rainfall,Evaporation,Sunshine,WindGustDir,WindGustSpeed,WindDir9am,...,Humidity9am,Humidity3pm,Pressure9am,Pressure3pm,Cloud9am,Cloud3pm,Temp9am,Temp3pm,RainToday,RainTomorrow
0,2008-12-01,Albury,13.4,22.9,0.6,NaN,NaN,W,44.0,W,...,71.0,22.0,1007.7,1007.1,8.0,NaN,16.9,21.8,No,No
1,2008-12-02,Albury,7.4,25.1,0.0,NaN,NaN,WNW,44.0,NNW,...,44.0,25.0,1010.6,1007.8,NaN,NaN,17.2,24.3,No,No
2,2008-12-03,Albury,12.9,25.7,0.0,NaN,NaN,WSW,46.0,W,...,38.0,30.0,1007.6,1008.7,NaN,2.0,21.0,23.2,No,No
3,2008-12-04,Albury,9.2,28.0,0.0,NaN,NaN,NE,24.0,SE,...,45.0,16.0,1017.6,1012.8,NaN,NaN,18.1,26.5,No,No
4,2008-12-05,Albury,17.5,32.3,1.0,NaN,NaN,W,41.0,ENE,...,82.0,33.0,1010.8,1006.0,7.0,8.0,17.8,29.7,No,No


## 2. Revisión de tipos de variables
Analizamos el formato de cada columna para asegurar que los números sean tratados como números y las fechas como fechas.

In [123]:
# Ver el tipo de dato de cada columna
df.dtypes

Date                 str
Location             str
MinTemp          float64
MaxTemp          float64
Rainfall         float64
Evaporation      float64
Sunshine         float64
WindGustDir          str
WindGustSpeed    float64
WindDir9am           str
WindDir3pm           str
WindSpeed9am     float64
WindSpeed3pm     float64
Humidity9am      float64
Humidity3pm      float64
Pressure9am      float64
Pressure3pm      float64
Cloud9am         float64
Cloud3pm         float64
Temp9am          float64
Temp3pm          float64
RainToday            str
RainTomorrow         str
dtype: object

## 3. Identificación de valores faltantes y duplicados
Analizamos la calidad de los datos detectando celdas vacías (valores nulos) y posibles filas duplicadas que puedan sesgar el análisis.

In [124]:
# Contar valores nulos (NaN) por columna
print("Valores nulos por columna:")
print(df.isnull().sum())

# Contar filas duplicadas
print("\nCantidad de filas duplicadas:", df.duplicated().sum())

Valores nulos por columna:
Date                 0
Location             0
MinTemp           1485
MaxTemp           1261
Rainfall          3261
Evaporation      62790
Sunshine         69835
WindGustDir      10326
WindGustSpeed    10263
WindDir9am       10566
WindDir3pm        4228
WindSpeed9am      1767
WindSpeed3pm      3062
Humidity9am       2654
Humidity3pm       4507
Pressure9am      15065
Pressure3pm      15028
Cloud9am         55888
Cloud3pm         59358
Temp9am           1767
Temp3pm           3609
RainToday         3261
RainTomorrow      3267
dtype: int64

Cantidad de filas duplicadas: 0


## 4. Limpieza de datos
En este paso, procedemos a eliminar registros duplicados para asegurar la integridad de la muestra y tratamos los valores nulos para evitar errores en los cálculos estadísticos.

In [125]:
# 1. Eliminar filas duplicadas
df.drop_duplicates(inplace=True)

# 2. Rellenar nulos de columnas numéricas con la mediana
# Esto mantiene la integridad de los datos sin perder filas
for col in df.select_dtypes(include=['float64']).columns:
    df[col] = df[col].fillna(df[col].median())

print("Limpieza completada: Se eliminaron duplicados y se imputaron valores faltantes.")

Limpieza completada: Se eliminaron duplicados y se imputaron valores faltantes.


## 5. Descripción inicial del dataset
Tras la limpieza, generamos un resumen estadístico que describe las principales métricas del dataset, permitiéndonos entender la distribución de las variables meteorológicas.

In [126]:
# Generar resumen estadístico de las variables numéricas
df.describe()

,MinTemp,MaxTemp,Rainfall,Evaporation,Sunshine,WindGustSpeed,WindSpeed9am,WindSpeed3pm,Humidity9am,Humidity3pm,Pressure9am,Pressure3pm,Cloud9am,Cloud3pm,Temp9am,Temp3pm
count,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000,145460.000000
mean,12.192053,23.215962,2.307990,5.179779,7.989889,39.962189,14.030751,18.669758,68.901251,51.553396,1017.644768,1015.250115,4.659755,4.709913,16.987101,21.668916
std,6.365780,7.088358,8.389771,3.178819,2.757790,13.120931,8.861796,8.716716,18.855360,20.471345,6.728484,6.663994,2.281490,2.106768,6.449299,6.850658
min,-8.500000,-4.800000,0.000000,0.000000,0.000000,6.000000,0.000000,0.000000,0.000000,0.000000,980.500000,977.100000,0.000000,0.000000,-7.200000,-5.400000
25%,7.700000,18.000000,0.000000,4.000000,8.200000,31.000000,7.000000,13.000000,57.000000,37.000000,1013.500000,1011.100000,3.000000,4.000000,12.300000,16.700000
50%,12.000000,22.600000,0.000000,4.800000,8.400000,39.000000,13.000000,19.000000,70.000000,52.000000,1017.600000,1015.200000,5.000000,5.000000,16.700000,21.100000
75%,16.800000,28.200000,0.600000,5.200000,8.700000,46.000000,19.000000,24.000000,83.000000,65.000000,1021.800000,1019.400000,6.000000,6.000000,21.500000,26.200000
max,33.900000,48.100000,371.000000,145.000000,14.500000,135.000000,130.000000,87.000000,100.000000,100.000000,1041.000000,1039.600000,9.000000,9.000000,40.200000,46.700000


Interpretación del Resumen Estadístico:

count: Indica que contamos con 145,460 registros válidos.

mean: Es el promedio de cada variable meteorológica.

std: (Desviación estándar) Mide la dispersión de los datos respecto al promedio.

min/max: Muestran los valores extremos registrados.

Cuartiles (25%, 50%, 75%): Dividen la distribución, permitiéndonos identificar el comportamiento central de las variables.

In [127]:
# Verificación final
print(f"Dimensiones finales del dataset: {df.shape}")
print("¿Quedan valores nulos?:", df.isnull().sum().sum() > 0)

Dimensiones finales del dataset: (145460, 23)
¿Quedan valores nulos?: True


In [128]:
# Verifica cuántos nulos hay por columna
nulos_por_columna = df.isnull().sum()
print(nulos_por_columna)

Date                 0
Location             0
MinTemp              0
MaxTemp              0
Rainfall             0
Evaporation          0
Sunshine             0
WindGustDir      10326
WindGustSpeed        0
WindDir9am       10566
WindDir3pm        4228
WindSpeed9am         0
WindSpeed3pm         0
Humidity9am          0
Humidity3pm          0
Pressure9am          0
Pressure3pm          0
Cloud9am             0
Cloud3pm             0
Temp9am              0
Temp3pm              0
RainToday         3261
RainTomorrow      3267
dtype: int64


Se realizó un conteo de valores faltantes (NaN) por cada variable. Los resultados muestran la presencia de celdas sin información, lo cual es una característica esperada en datasets meteorológicos de gran escala. Estos nulos han sido identificados para su gestión estratégica en las etapas posteriores del proyecto.

In [129]:
# Limpieza de variables categóricas
cat_cols = ['WindGustDir', 'WindDir9am', 'WindDir3pm', 'RainToday', 'RainTomorrow']

for col in cat_cols:
    # Rellenamos los nulos con la moda (el valor más común)
    df[col] = df[col].fillna(df[col].mode()[0])

print("Limpieza de variables categóricas completada.")

Limpieza de variables categóricas completada.


Se identificó que las variables categóricas presentaban valores nulos que no fueron abordados por la imputación numérica basada en la mediana. Se aplicó una imputación basada en la moda (valor más frecuente) para estas variables, asegurando así la completitud del dataset sin alterar la distribución de las categorías, permitiendo que el conjunto de datos esté íntegramente disponible para las siguientes etapas de modelamiento predictivo.

In [131]:
# Limpieza forzada y profesional de columnas de dirección del viento
cols_direccion = ['WindGustDir', 'WindDir9am', 'WindDir3pm']

for col in cols_direccion:
    # Calculamos la moda
    moda = df[col].mode()[0]
    # Usamos loc para asegurar que estamos modificando el DataFrame original sin ambigüedades
    df.loc[:, col] = df[col].fillna(moda)

# Verificación final
print("Limpieza finalizada con éxito. Nulos restantes en direcciones:", df[cols_direccion].isnull().sum().sum())

Limpieza finalizada con éxito. Nulos restantes en direcciones: 0
